libbeat/monitoring/inputmon/httphandler.go

```diff
@@ -142,16 +142,16 @@ func addInputMetrics(to map[string]inputMetricsTable, from map[string]any, pathP
 		}
 
 		inputPath := append(pathPrefix, key)
+		to[id] = inputMetricsTable{
```

Comment:
This was the root cause of the crash. The ids from registries tagged as "nested" were skipped in collision checks (but winlog used the same id for its container registry _and_ its main data source). Including ids unconditionally fixes it.